chia/util/merkle_set.py

```diff
@@ -142,6 +141,12 @@ def _audit(self, hashes: List[bytes]) -> None:
         self.root._audit(newhashes, [])
         assert newhashes == sorted(newhashes)
 
+    @staticmethod
+    def _with_root(root: Node) -> MerkleSet:
+        s = MerkleSet([])
+        s.root = root
+        return s
```

Comment:
Feels awkward with it being private but used outside the class.  Also feels like an awkward override just assigning to the attribute to throw away whatever work the constructor does.  Also also relates to some comments I'll submit on https://github.com/Chia-Network/chia-blockchain/pull/17867.  Also also also, clearly not a huge deal.  Also also also also, kinda counter to the 'making it immutable' title.

Reply:
My understanding of the `@overload` is that it's just a typing annotation, you still only have one implementation of `__init__()`, right?
The `deserialize_proof()` function really should be part of `MerkleSet` so I consider it private. It should be another constructor.

Reply:
Yes, `@overload` is just type hinting.  It is a way to clearly express the two intended uses of a function with (if you are referring to my comments around `.__init__()`) two mutually exclusive parameters.  Now sure, they technically aren't exclusive in the sense that you _can_ pass both and it will work.  But it is kind of an odd use of the function given that one of the parameters is ignored.  (except not anymore since you removed the `root=` parameter)

Given that the target is to delete this code as we shift it to rust, I don't need to discuss this anymore.  Feel free to resolve this discussion.  Or, maybe keybase chat if you are interested in trying to get me to flesh out a pattern here that I would like.